Model/Api/OrderManagement.php

```diff
@@ -189,6 +189,33 @@ public function manage(
         }
     }
 
+    /**
+     * Handle custom checkboxes if request has them
+     *
+     * @param string $displayId
+     */
+    private function handleCheckboxes($displayId) {
```

Comment:
For extensibility / customization purpose we should at lest make this method public so we can write plugins (M2 interceptors) for it on the fly. This can be some sort of a dispatcher to methods that process specific checkbox categories. We should do the processing of every checkbox category out of this method. Maybe we should think of some naming convention so they can be automatically called if exist and we put them in a separate, extendable file.
@matt-thomason @njfaries @YevhenShBolt @johnmccombs1 thoughts?

Reply:
Great idea @danac-gs. I don't have a preference for any naming convention - open to suggestions.

Reply:
Also agree that method must be a public.

Reply:
Changed to public. @danac-gs any ideas about naming convention?

Reply:
handleCheckbox<Checkboxname>,  handleCheckboxNewsletter for instance.
And put the handlers in a separate class, eg. Helper\CheckboxHandlers.
That way we can override / extend that file in a separate, extension package (uploaded to app/code/Bolt/Bolt) when a customization is required (custom checkbox) without affecting the code plugin code.

Reply:
@danac-gs, For now, it seems logical to handle all checkboxes in one method, because we create only one comment with info of all checkboxes.
My proposal is just moving most part of handleCheckboxes into a separate helper class, like **handleCheckboxes($checkboxes)**.
If down the road we see that it's not enough for real integration tasks, we can improve it.

Reply:
Well, we can start with that. Then override that method or the whole helper class in a custom extension when needed.
What I was thinking of is a thin method that just loops through checkboxes and calls a handler for every one of them, passing it (all) the data available - the checkbox record from transaction, quote and order. This way, when dealing with custom checkboxes, we don't change the main handler, just add the handler for specific checkbox.
Adding order history comments regarding checkboxes can / should stay in the parent handler.
If we expect this custom checkboxes to be really custom then we should decouple their implementation handlers. If not, if subscribeForNewsletter would be the only one, then I got it all wrong (confused by the word custom) and we can leave it as is.

Reply:
Yeah, I got what you offered, it's a good solution, but maybe it's better to wait for real cases. So just moving handleCheckboxes to separate helper for now.